pyeudiw/satosa/frontends/openid4vci/endpoints/authorization_endpoint.py

```diff
@@ -51,46 +62,119 @@ def endpoint(self, context: Context) -> Response:
             if context.request_method == "POST":
+                if not context.http_headers or HTTP_CONTENT_TYPE_HEADER not in context.http_headers:
+                    self._log_error(
+                        AUTHORIZATION_ENDPOINT,
+                        "missing content-type header"
+                    )
+                    raise InvalidRequestException("invalid content-type")
+                
                 validate_content_type(context.http_headers[HTTP_CONTENT_TYPE_HEADER], FORM_URLENCODED)
                 auth_req = self._get_body(context)
             else:
-                validate_content_type(context.http_headers[HTTP_CONTENT_TYPE_HEADER], APPLICATION_JSON)
                 auth_req = context.qs_params
 
             if not auth_req:
                 raise InvalidRequestException("missing authorization request")
+            
+            req_uri = auth_req.get("request_uri", None)
+
+            if not req_uri:
+                raise InvalidRequestException("missing request_uri in authorization request")
+
+            entity = self.db_engine.search_session_by_field("request_uri_part", req_uri.split(":")[-1])
+
+            if not entity:
+                raise InvalidRequestException(f"request_uri `{req_uri}` not found in storage")
+
+            vci_entity = OpenId4VCIEntity(**entity)
+
+            old_session_id = vci_entity.session_id
+
+            vci_entity.session_id = get_session_id(context)
+            self.db_engine.upsert_session(old_session_id, vci_entity.model_dump())
 
             AuthorizationRequest.model_validate(
                 auth_req, context = {
                     ENDPOINT_CTX: AUTHORIZATION_ENDPOINT,
-                    PAR_REQUEST_URI_CTX: self._to_request_uri(entity.request_uri_part),
-                    CLIENT_ID_CTX: entity.client_id
+                    PAR_REQUEST_URI_CTX: self._to_request_uri(entity["request_uri_part"]),
+                    CLIENT_ID_CTX: vci_entity.client_id
                 })
-            return AuthorizationResponse(
-                state=entity.state,
-                iss=self.entity_id,
-            ).to_redirect_response(entity.redirect_uri)
+            
+            if not self._auth_callback:
+                raise Exception("missing auth_callback for authorization endpoint")
+            
+            internal_req = InternalData(
+                subject_type="pairwise",
+                requester=vci_entity.client_id,
+                requester_name=None,
+            )
+
+            if not self._converter:
+                raise Exception("missing attribute converter for authorization endpoint")
+
+            # TODO: handle attributes based on the backend
+            internal_req.attributes = self._converter.to_internal_filter(
+                "openid",
+                {}
+            )
+
+            context.target_backend = self.config.get("default_target_backend", "spidSaml2")
```

Comment:
```suggestion
            context.target_backend = self.config.get("default_target_backend", "spidSaml2")
```

please don't put constants like this ... default target backed is something configured within the proxy, not intended to be stucked in a frontend code